noir_stdlib/src/cmp.nr

```diff
@@ -213,6 +223,26 @@ impl<T, N> Ord for [T; N] where T: Ord {
     }
 }
 
+impl<T> Ord for [T] where T: Ord {
+    // The first non-equal element of both arrays determines
+    // the ordering for the whole array.
+    fn cmp(self, other: [T]) -> Ordering {
+        let mut result = Ordering::equal();
+        for i in 0 .. self.len() {
+            if result == Ordering::equal() {
```

Comment:
Cmp for slices also needs to account for the slices possibly being different lengths